### PR TITLE
Upgrade k8s client dependencies

### PR DESCRIPTION
**Pull request by lblackstone**
### Proposed changes
- Upgrade helm to v3.9.0
- Upgrade k8s client-go to v0.24.0
- Fix client usage where methods have changed. The upstream discovery client now includes caching, so remove our forked implementation and use the upstream method directly.

### CODE REVIEW

**Comment by lblackstone** [comment]
@viveklak The most interesting part of this change is the removal of our forked memcached discovery client. The upstream library now implements caching, and I verified that the tests took approximately the same amount of time to run, so this doesn't appear to regress performance (as expected).

File: provider/pkg/clients/memcache.go [hunk 1]
```diff
@@ -1,254 +0,0 @@
...
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-//
-// Note: this code is taken nearly verbatim from the upstream memcache client [1].
-// The `refreshLocked` method does not return errors, but instead logs them using
-// the `HandleError` method, which dispatches to klog. This bypasses Pulumi's logging
-// mechanism, and causes unactionable error messages to appear in the CLI.
-// This error logging was disabled.
```
Comment (viveklak) [inline provider/pkg/clients/memcache.go:24]: Any thoughts on the logging here? Is this still relevant? Would it affect users?

**Review by viveklak — APPROVED** [review]
Reminder to add a changelog to document bump of helm and k8s client deps. Also left one comment regarding the cache client change.

File: provider/pkg/clients/memcache.go (hunk 1 above)
Comment (lblackstone) [reply provider/pkg/clients/memcache.go]: Updated. Good catch!

**Pull request merged**